### PR TITLE
Inputs

### PR DESCRIPTION
**Pull request by frederikstroem**

### CODE REVIEW

**Review by Nicklason — APPROVED** [review]

**Review by Nicklason — APPROVED** [review]

**Review by Kjalris — COMMENTED** [review]
Add more input keys such as enter (to enter shop)
R for reloading a gun
Mouse1 for clicking
Mouse2 for clicking 
This needs to be added in both the ENUM class and the keyboardinput

**Review by Kjalris — APPROVED** [review]
The core cant run because of the texture. Need to out comment the texture in enemy (the absolute path)
Also in Core, both the batch with the absolute path and inside the render() function, the 2nd batch with the map needs to be out commented

**Comment by Nicklason** [comment]
> The core cant run because of the texture. Need to out comment the texture in enemy (the absolute path)
> Also in Core, both the batch with the absolute path and inside the render() function, the 2nd batch with the map needs to be out commented

I have made an issue for it https://github.com/Nicklason/4-semester-project/issues/44

File: dk.sdu.se4.keyboardinput/src/main/java/dk/sdu/se4/keyboardinput/KeyboardInputImpl.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    @Override
+    public boolean keyDown(int keycode) {
+        //System.out.println("DOWN " + Keys.toString(keycode));
+        switch (keycode) {
```
Comment (stped18) [inline dk.sdu.se4.keyboardinput/src/main/java/dk/sdu/se4/keyboardinput/KeyboardInputImpl.java:24]: Dont use switch statements

File: dk.sdu.se4.keyboardinput/src/main/java/dk/sdu/se4/keyboardinput/KeyboardInputImpl.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+                break;
+        }
+        
+        return true;
+    }
+    
+    @Override
+    public boolean keyUp(int keycode) {
+        //System.out.println("UP " + Keys.toString(keycode));
+        switch (keycode) {
```
Comment (stped18) [inline dk.sdu.se4.keyboardinput/src/main/java/dk/sdu/se4/keyboardinput/KeyboardInputImpl.java:51]: dont use switch statements

File: dk.sdu.se4.keyboardinput/src/main/java/dk/sdu/se4/keyboardinput/KeyboardInputImpl.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+import dk.sdu.se4.commongameinput.GameInputKeys;
+import java.util.HashMap;
+import java.util.Map;
+import com.badlogic.gdx.Input.Keys;
+import com.badlogic.gdx.InputAdapter;
+
+public class KeyboardInputImpl extends InputAdapter implements GameInput {
+    private final Map<GameInputKeys, Boolean> keys = new HashMap();
+    
+    public boolean isPressed(GameInputKeys key) {
```
Comment (stped18) [inline dk.sdu.se4.keyboardinput/src/main/java/dk/sdu/se4/keyboardinput/KeyboardInputImpl.java:13]: keys.get(key) this returns an boolean ?

**Review by stped18 — CHANGES_REQUESTED** [review]

File: dk.sdu.se4.keyboardinput/src/main/java/dk/sdu/se4/keyboardinput/KeyboardInputImpl.java (hunk 1 above)
Comment (Nicklason) [reply dk.sdu.se4.keyboardinput/src/main/java/dk/sdu/se4/keyboardinput/KeyboardInputImpl.java:24]: Why not?

File: dk.sdu.se4.keyboardinput/src/main/java/dk/sdu/se4/keyboardinput/KeyboardInputImpl.java (hunk 3 above)
Comment (Nicklason) [reply dk.sdu.se4.keyboardinput/src/main/java/dk/sdu/se4/keyboardinput/KeyboardInputImpl.java:13]: Or null if the key is not saved

File: dk.sdu.se4.keyboardinput/src/main/java/dk/sdu/se4/keyboardinput/KeyboardInputImpl.java (hunk 1 above)
Comment (stped18) [reply dk.sdu.se4.keyboardinput/src/main/java/dk/sdu/se4/keyboardinput/KeyboardInputImpl.java:24]: code smell

Comment (stped18) [reply dk.sdu.se4.keyboardinput/src/main/java/dk/sdu/se4/keyboardinput/KeyboardInputImpl.java:24]: en måde at løse det på er at lave det om til en liste af keys og validate dem

**Review by stped18 — APPROVED** [review]

**Pull request merged**